plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/AttributeMorpher.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2018-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.conda.internal.hosted;
+
+import groovy.lang.Tuple2;
+import org.apache.commons.lang.StringUtils;
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+import org.sonatype.nexus.repository.conda.internal.hosted.metadata.PackageDesc;
+import org.sonatype.nexus.repository.storage.Asset;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+
+public class AttributeMorpher {
+
+    public static Tuple2<String, PackageDesc> toPackageDesc(Asset asset) {
+
+        NestedAttributesMap attributes = asset.formatAttributes();
+        PackageDesc packageDesc = new PackageDesc();
+
+        packageDesc.setSize(asset.size().longValue());
+        packageDesc.setMd5(asset.getChecksum(HashAlgorithm.MD5).toString());
+        packageDesc.setSha256(asset.getChecksum(HashAlgorithm.SHA256).toString());
+        packageDesc.setTimestamp(Long.parseLong(attributes.get("timestamp", 0).toString()));
```

Comment:
Let's create constants for attribute field names

Reply:
NestedAttributesMap can accept the actual object so all those strings will not be needed - added TODO info